### DISCUSSION TITLE
Validation of model instances

### DISCUSSION THREAD

**Discussion by popovicieduard** [adonisjs/core#2081] [Help]
Hello,

I was wondering if there is a way to validate a instance of a model inside a validator without using the object method, for example:

```
import User from 'App/Models/User'

schema.create({
    user: schema.object().anyMembers(), // --> something like schema.instanceOf(User)
  })
```

I know I can create a custom validation rule but how should I approach this? Is lazy loading the class inside the validation rule a feasible approach? Or should I stick to the object method?

**Comment by thetutlage** [comment]
Can you share how you are planning to use this? The validator is mostly used to validate the HTTP requests and you cannot get class instances via HTTP request

**Reply by popovicieduard** [reply]
I am using the validator standalone package in my methods, so for example:

```
import { validator } from '@ioc:Adonis/Core/Validator'

public async create(params: fooContract) {
  const fooValidator= new FooValidator()

  const validatedParams = await validator.validate({
    schema: fooValidator.schema,
    data: params, //=> here the params.user is a User instance, so I want to validate it, I don't know if it's also possible to have the type passed back to validatedParams, so validatedParams.user becomes of type User
    messages: fooValidator.messages,
  })

  const user = params.user // if I replace this with validatedParams.user, user loses type

  const foo= await user.related('foo').create({...validatedParams})

  return foo
}
```
